HW1/P4/P4.ipynb

```diff
@@ -0,0 +1,1206 @@
+{
```

Comment:
Please submit .py files instead of ipython notebooks.  It makes it much harder to parse and read when examining pull requests!